### DISCUSSION TITLE
Automatically retry after failed upload in synchronize?

### DISCUSSION THREAD

**Discussion by gctypo** [abraunegg/onedrive#1731] [Q&A] [answered]
Is there a way to get a `--synchronize` operation to not abort when an upload fails, and instead retry until successful? I have really shoddy internet that causes a sync to return 141 every time it blips out for a minute. Is there a way to do this within the client, or do I just need to write a shell script to rerun the command after a delay if it fails?

**Comment by abraunegg** [comment, marked as answer]
@gctypo 
The `--synchronize` is meant for a one-off sync and exit.

You could potentially use `--monitor` via a systemd service to constantly keep the service running & retry when there is a disruption, but your best options for resolution are:
* Write a shell script for your purpose
* Change ISP to a more reliable service

**Reply by gctypo** [reply]
Changing ISP is not an option because monopoly, quick and dirty shell script to auto-retry it is.
